samples/constitutions/trusted_authority/resolve.js

```diff
@@ -0,0 +1,78 @@
+function getMemberInfo(memberId) {
+  return ccf.bufToJsonCompatible(
+    ccf.kv["public:ccf.gov.members.info"].get(ccf.strToBuf(memberId))
+  );
+}
+
+function isRecoveryMember(memberId) {
+  return (
+    ccf.kv["public:ccf.gov.members.encryption_public_keys"].get(
+      ccf.strToBuf(memberId)
+    ) ?? false
+  );
+}
+
+// Defines which of the members are operators.
+function isOperator(memberId) {
+  // Operators cannot be recovery members.
+  if (isRecoveryMember(memberId)) {
+    return false;
+  }
+  return getMemberInfo(memberId).member_data?.is_operator ?? false;
+}
+
+// Defines which of the members are operator provisioners.
+function isOperatorProvisioner(memberId) {
+  return (
+    !isRecoveryMember(memberId) &&
+    (getMemberInfo(memberId).member_data?.is_operator_provisioner ?? false)
+  );
+}
+
+// Defines actions that can be passed with sole operator provisioner input.
+function canOperatorProvisionerPass(action) {
+  // Operator provisioners can add or retire operators.
+  return (
+    {
+      set_member: () => action.args["member_data"]?.is_operator ?? false,
+      remove_member: () => isOperator(action.args.memberId),
+    }[action.name.toString()]() ?? false
+  );
+}
+
+export function resolve(proposal, proposerId, votes) {
+  const actions = JSON.parse(proposal)["actions"];
+
+  // If the node is an operator provisioner, strictly enforce what proposals it can
+  // make
+  if (isOperatorProvisioner(proposer_id)) {
+    return actions.every(canOperatorProvisionerPass) ? "Accepted" : "Rejected";
```

Comment:
It could be tempting to be paranoid about this, and check that the proposal contains exactly one of each.

I wonder in particular what would happen with:

```
set_member foo
remove_member bar
remove_member foo
```

In a system with an operator bar to start with. I suspect the net result is a system without an operator, which doesn't seem ideal!

Reply:
Are we thinking that the provisioner is only for the disaster recovery? Would it be responsible for provisioning the very first operator (in which case I guess we would need to allow either just an add and maybe just a remove?)

We could certainly enforce provisioners can only do one in, one out actions, but if not, I guess we can't prevent a provisioner removing a member it didn't set it?

Reply:
The initial provisioning is done through config, and effectively signed off by the service open proposal.